gdal_subsetter/utilities.py

```diff
@@ -0,0 +1,25 @@
+""" A module to contain helper functions offering utility functionality for
+    the Harmony GDAL Adapter Service.
+
+"""
+from os.path import exists as file_exists, splitext
+
+
+known_file_types = {'.nc': 'nc', '.nc4': 'nc', '.tif': 'tif', '.tiff': 'tif',
+                    '.zip': 'zip'}
+
+
+def get_file_type(input_file_name: str) -> str:
+    """ Determine the input file type according to its extension. If the
+        format is not recognised the extension is returned so it can be
+        provided in a raised exception and the message returned to the
+        end-user.
+
+    """
+    if input_file_name is None or not file_exists(input_file_name):
+        file_type = None
```

Comment:
Perhaps when doing further refactoring, this function could be updated to raise an exception here. I didn't want to tweak things too much at first, though.